service/s3/endpoints.go

```diff
@@ -4325,3 +4239,602 @@ func (r *resolver) ResolveEndpoint(
 	}
 	return endpoint, fmt.Errorf("endpoint rule error, %s", "A region must be set when sending requests to S3.")
 }
+
+type endpointParamsBinder interface {
+	bindEndpointParams(*EndpointParameters)
+}
+
+func bindEndpointParams(input endpointParamsBinder, options Options) *EndpointParameters {
+	params := &EndpointParameters{}
+
+	params.Region = aws.String(endpoints.MapFIPSRegion(options.Region))
+	params.UseFIPS = aws.Bool(options.EndpointOptions.UseFIPSEndpoint == aws.FIPSEndpointStateEnabled)
+	params.UseDualStack = aws.Bool(options.EndpointOptions.UseDualStackEndpoint == aws.DualStackEndpointStateEnabled)
+	params.Endpoint = options.BaseEndpoint
+	params.ForcePathStyle = aws.Bool(options.UsePathStyle)
+	params.Accelerate = aws.Bool(options.UseAccelerate)
+	params.DisableMultiRegionAccessPoints = aws.Bool(options.DisableMultiRegionAccessPoints)
+	params.UseArnRegion = aws.Bool(options.UseARNRegion)
+
+	input.bindEndpointParams(params)
+
+	return params
+}
+
+func (in *AbortMultipartUploadInput) bindEndpointParams(p *EndpointParameters) {
```

Comment:
just curious, was there a strong reason to put all the operation input receivers for `bindEndpointParams` here rather than in the operation specfic files where the operation input struct is defined? or was it more of a toss up between "it being endpoints related functionality" and "it being an operation method"?

Reply:
Moved after discussing offline.